lib/connection.js

```diff
@@ -165,18 +165,22 @@ Connection.prototype.sendResult = function sendResult (id, error, result) {
     Assert(error || result, 'Must have an error or a result.');
     Assert( !( error && result ), 'Cannot have both an error and a result');
 
-    this.sendRaw({
-        id: id,
-        result: result,
-        error: error
-    });
+    var response = { id: id };
+
+    if (result) {
```

Comment:
Technically the assert above was ensuring that only error or result existed, and `JSON.stringify` omits undefined.

```
> JSON.stringify({a: 1, b: undefined});
'{"a":1}'
```

I see now though that this may have been too 'clever' an approach, and yours is much more readable and also prevents the possibility of a `null` ending up in there.